### PR TITLE
fixes to make CUDA work and MPQC compile

### PR DESCRIPTION
**Pull request by evaleev**
also includes MADNESS fixes that prevent most common case of dangling refs in `array.find().get()`

### CODE REVIEW

File: src/TiledArray/dist_array.h [hunk 1]
```diff
@@ -544,6 +544,59 @@ class DistArray : public madness::archive::ParallelSerializableObject {
     return find<std::initializer_list<Integer>>(i);
   }
 
+  /// Find local tile
+
+  /// \tparam Index An integral or integral range type
+  /// \param i The tile index
+  /// \return A \c future to tile \c i
+  /// \throw TiledArray::Exception When tile \c i is zero or not local
+  template <typename Index,
```
Comment (ryanmrichard) [inline src/TiledArray/dist_array.h:553]: This should use the `enable_if_is_integral_or_integral_range` type to be consistent with the rest of the class.

File: src/TiledArray/dist_array.h [hunk 2]
```diff
@@ -544,6 +544,59 @@ class DistArray : public madness::archive::ParallelSerializableObject {
...
+    return pimpl_->get_local(i);
+  }
+
+  /// Find local tile
+
+  /// \tparam Integer An integral type
+  /// \param i The tile index, as an \c std::initializer_list<Integer>
+  /// \return A \c future to tile \c i
+  /// \throw TiledArray::Exception When tile \c i is zero or not local
+  template <typename Integer,
```
Comment (ryanmrichard) [inline src/TiledArray/dist_array.h:567]: similarly use the `enable_if_is_integral`

File: src/TiledArray/dist_array.h [hunk 3]
```diff
@@ -544,6 +544,59 @@ class DistArray : public madness::archive::ParallelSerializableObject {
...
+  }
+
+  /// Find local tile
+
+  /// \tparam Index An integral or integral range type
+  /// \param i The tile index
+  /// \return A \c future to tile \c i
+  /// \throw TiledArray::Exception When tile \c i is zero or not local
+  template <typename Index,
+            typename = std::enable_if_t<std::is_integral_v<Index> ||
```
Comment (ryanmrichard) [inline src/TiledArray/dist_array.h:581]: ditto

**Comment by evaleev** [comment]
Yes, unfortunately now TA changes can/do break mpqc so it is likely that
this PR will break something in NWX.

**Pull request merged**